ee/codegen/src/project.ts

```diff
@@ -421,88 +399,85 @@ ${errors.slice(0, 3).map((err) => {
   }
 
   /**
-   * This method is used to order the nodes based on a declared import order, determined by a
-   * breadth first search over edges in the graph.
+   * This method is used to order the nodes based on dependencies between them.
+   * It ensures that nodes appear in the result after all the nodes they depend on.
    */
   private getOrderedNodes(): WorkflowDataNode[] {
     const rawData = this.workflowVersionExecConfig.workflowRawData;
+    const nodes = rawData.nodes.filter(
+      (node): node is WorkflowDataNode => node.type !== "ENTRYPOINT"
+    );
 
     // Edge case: Workflow init only has two nodes of ENTRYPOINT and TERMINAL with no edge between them
     if (rawData.edges.length === 0) {
-      return rawData.nodes.filter(
-        (node): node is WorkflowDataNode => node.type !== "ENTRYPOINT"
-      );
+      return nodes;
     }
 
-    const nodesById = Object.fromEntries(
-      rawData.nodes.map((node) => [node.id, node])
-    );
+    const nodesById = Object.fromEntries(nodes.map((node) => [node.id, node]));
 
-    const edgesQueue = this.workflowContext.getEntrypointNodeEdges();
-    const edgesByPortId = this.workflowContext.getEdgesByPortId();
-    const processedEdges = new Set<WorkflowEdge>();
-    const processedNodeIds = new Set<string>();
+    // Create a dependency graph that represents the "depends on" relationships between nodes
+    // If A depends on B, then B must come before A in the result
+    const dependencyGraph: Map<string, Set<string>> = new Map();
 
-    const orderedNodes: WorkflowDataNode[] = [];
+    // Initialize dependency graph with all nodes
+    nodes.forEach((node) => {
+      dependencyGraph.set(node.id, new Set());
+    });
 
-    while (edgesQueue.length > 0) {
-      const edge = edgesQueue.shift();
-      if (!edge) {
-        continue;
-      }
+    // Build dependency relationships based on edges
+    rawData.edges.forEach((edge) => {
+      const sourceNodeId = edge.sourceNodeId;
+      const targetNodeId = edge.targetNodeId;
 
-      const sourceNode = nodesById[edge.sourceNodeId];
-      if (!sourceNode) {
-        continue;
+      const targetDependencies = dependencyGraph.get(targetNodeId);
+      const sourceDependencies = dependencyGraph.get(sourceNodeId);
+      if (targetDependencies && sourceDependencies) {
```

Comment:
[nit] do we need to check for the existence of `sourceDependencies`? but it's not used in the block